### PR TITLE
Add Script that Detects Changes required for Platform Compatibility Tool

### PR DESCRIPTION
**Pull request by nimanch**
This is Step 1 of a 2 Step PR that runs a script to deploy IoT Edge Binaries and Containers to a device and then detects whether recorded Binary Size, Memory Usage in the platform compatibility tool is not less than or greater than the current runs Memory Usage, Binary Size (A Buffer of 50MB is provided)

The script will error out if there is a difference and prompt to update the compatibility tool. I am not choosing to auto update the tool because a significant difference in memory/size should be investigated before updating the compatibility tool

TODO:
1. Add Additional checks - Shared Library , Docker Version
2. Add a YAML for Scheduled CI Builds for ARM32, ARM64 and AMD64 

***Please replace this line with your PR description and read PR checklist below***

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: platform-validation/scripts/aziot-compatibility.sh [hunk 1]
```diff
@@ -605,21 +605,21 @@ check_package_manager() {
...
     for package in $package_managers; do
-    {
-        res="$(need_cmd $package)"  
-        if [ $? -eq 0 ] ; then
-            not_found=0
-            wrap_debug "Current target platform supports $package package manager"
-            wrap_pass "check_package_manager"
-            if [ $package = "rpm" ] || [ $package = "dpkg" ]; then
-                check_ca_cert
+        {
```
Comment (ggjjj) [inline platform-validation/scripts/aziot-compatibility.sh:608]: it got intended by mistake?

Comment (nimanch) [reply platform-validation/scripts/aziot-compatibility.sh]: I removed the curly braces, what I have seen in shell/bash scripts is that indentation alone is enough.

File: platform-validation/deployment/platform-compatibility.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{
+    "modulesContent": {
+        "$edgeAgent": {
+            "properties.desired": {
+                "modules": {
+                    "SimulatedTemperatureSensor": {
```
Comment (darobs) [inline platform-validation/deployment/platform-compatibility.json:6]: What's the advantage to running the simulated temp sensor?

**Review by darobs — DISMISSED** [review]

File: platform-validation/scripts/aziot-compatibility.sh [hunk 3]
```diff
@@ -634,19 +634,22 @@ check_package_manager() {
...
     package_managers="apt-get dnf yum dpkg rpm"
     for package in $package_managers; do
-        res="$(need_cmd $package)"
-        if [ $? -eq 0 ]; then
-            not_found=0
-            wrap_debug "Current target platform supports $package package manager"
-            wrap_pass "check_package_manager"
-            if [ $package = "rpm" ] || [ $package = "dpkg" ]; then
-                check_ca_cert
+        {
```
Comment (darobs) [inline platform-validation/scripts/aziot-compatibility.sh:637]: Looks like the braces came back.

**Review by darobs — APPROVED** [review]

**Pull request merged**